starter-code/routes/signin.js

```diff
@@ -0,0 +1,51 @@
+const express 	   = require("express");
+const bcrypt 	   = require("bcrypt");
+const userModel    = require('../model/user');
+const router       = express.Router();
+
+router.get('/signin',(req, res, next) => {
+  if (req.session.currentUser) {
```

Comment:
instead of writing this guard two times (in each of the routes in this router module), you could have added a middleware at the top - less repetition in your code

````
router.use(function (req, res, next) {
  if (req.session.currentUser) {
    next(); // we DO NOT have a user in the session, continue executing routes
  } else {
    res.redirect('/private'); // user is already logged in, redirect to /welcome
  }
});
```